src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -5052,13 +5031,9 @@ private void writeObject(ObjectOutputStream s) throws IOException {
      * Returns the mag array as an array of bytes.
      */
     private byte[] magSerializedForm() {
-        int len = mag.length;
-
-        int bitLen = (len == 0 ? 0 : ((len - 1) << 5) + bitLengthForInt(mag[0]));
-        int byteLen = (bitLen + 7) >>> 3;
-        byte[] result = new byte[byteLen];
+        byte[] result = new byte[(magBitLength() + 7) >>> 3];
```

Comment:
I think there's a risk of overflow, so consider
```suggestion
        byte[] result = new byte[(magBitLength() - 1 >>> 3) + 1];
```

Reply:
Ah no, there's a unsigned shift `>>>`.
Disregard

Reply:
Indeed, I'd thought about this possibility too.